src/raftstore/store/worker/apply.rs

```diff
@@ -1195,6 +1216,31 @@ impl ApplyDelegate {
 
         Ok(resp)
     }
+
+    fn handle_ingest_sst(
+        &mut self,
+        ctx: &ApplyContext,
+        req: &Request,
+        ssts: &mut Vec<SSTMeta>,
+    ) -> Result<Response> {
+        let sst = req.get_ingest_sst().get_sst();
+
+        if let Err(e) = check_sst_for_ingestion(sst, &self.region) {
+            error!("ingest {:?} to region {:?}: {:?}", sst, self.region, e);
+            // This file is not valid, we can delete it here.
+            let _ = ctx.importer.delete(sst);
+            return Err(e);
+        }
+
+        ctx.importer.ingest(sst, &self.engine).unwrap_or_else(|e| {
```

Comment:
Seems we don't verify checksum for this sst file?

Reply:
The checksum is verified inside `importer.ingest()`.